### PR TITLE
[`tests` / `Quantization`] Fix bnb test

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

Fixes the currently bnb failing test ([Link to failing job](https://github.com/huggingface/transformers/actions/runs/6680718296))

```
tests/quantization/bnb/test_mixed_int8.py::MixedInt8GPT2Test::test_get_keys_to_not_convert
```

The issue is related to the fact that currently the `trust_remote_code` version of MPT models cannot be loaded from transformers main since https://github.com/huggingface/transformers/pull/27086 removed some private methods from the modeling files, i.e. this simple script fails on main:

```python
from accelerate import init_empty_weights
from transformers import AutoModelForCausalLM, AutoConfig

model_id = "mosaicml/mpt-7b"
config = AutoConfig.from_pretrained(
    model_id, trust_remote_code=True, revision="72e5f594ce36f9cabfa2a9fd8f58b491eb467ee7"
)
with init_empty_weights():
    model = AutoModelForCausalLM.from_config(
        config, trust_remote_code=True, code_revision="72e5f594ce36f9cabfa2a9fd8f58b491eb467ee7"
    )
```

I raised an issue on the Hub https://huggingface.co/mosaicml/mpt-7b/discussions/83 so the issue should be hopefully fixed soon. I proposed to temporary skip the test for trust_remote_code version of MPT models and revert it back once that issue gets resolved. I believe that should be fine

### CODE REVIEW

File: tests/quantization/bnb/test_mixed_int8.py [hunk 1]
```diff
@@ -134,14 +134,16 @@ def test_get_keys_to_not_convert(self):
...
-        self.assertEqual(get_keys_to_not_convert(model), ["transformer.wte"])
+        # TODO: to revert once: https://huggingface.co/mosaicml/mpt-7b/discussions/83 is fixed
+        # config = AutoConfig.from_pretrained(
+        #     model_id, trust_remote_code=True, revision="72e5f594ce36f9cabfa2a9fd8f58b491eb467ee7"
+        # )
+        # with init_empty_weights():
+        #     model = AutoModelForCausalLM.from_config(
+        #         config, trust_remote_code=True, code_revision="72e5f594ce36f9cabfa2a9fd8f58b491eb467ee7"
+        #     )
+        # self.assertEqual(get_keys_to_not_convert(model), ["transformer.wte"])
```
Comment (amyeroberts) [inline tests/quantization/bnb/test_mixed_int8.py:145]: This should be split into a separate test e.g. `test_get_keys _to_not_convert_remote_code` and explicitly skipped with `unittest.skip` as well as an issue raised in this repo to track whether it's been resolved.

**Review by amyeroberts — COMMENTED** [review]
Thanks for handling!

Happy for this to be skipped for now, but it should be handled with proper skipping logic and tracked with a github issue

**Comment by younesbelkada** [comment]
Thanks! makes sense, adapted the test accordingly and made : https://github.com/mosaicml/llm-foundry/issues/703 to track the issue

**Review by amyeroberts — APPROVED** [review]
Thanks for iterating!

**Pull request merged**